influxdb3_write/src/paths.rs

```diff
@@ -0,0 +1,171 @@
+use crate::SegmentId;
+use chrono::prelude::*;
+use object_store::path::Path as ObjPath;
+use std::convert::AsRef;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// File extension for catalog files
+pub const CATALOG_FILE_EXTENSION: &str = "json";
+
+/// File extension for parquet files
+pub const PARQUET_FILE_EXTENSION: &str = "parquet";
+
+/// File extension for segment info files
+pub const SEGMENT_INFO_FILE_EXTENSION: &str = "info.json";
+
+/// File extension for segment wal files
+pub const SEGMENT_WAL_FILE_EXTENSION: &str = "wal";
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CatalogFilePath(ObjPath);
+
+impl CatalogFilePath {
+    pub fn new(segment_id: SegmentId) -> Self {
+        let path = ObjPath::from(format!(
+            "catalogs/{:010}.{}",
+            u32::MAX - segment_id.0,
```

Comment:
since this is a pattern that shows up elsewhere, maybe refactor into a method on SegmentId? object_store_file_stem?

Reply:
Implemented in [d1fbbe8](https://github.com/influxdata/influxdb/pull/24579/commits/d1fbbe8749cc4ff7aebf31d360f9e075001686a5)